lib/minify.js

```diff
@@ -178,13 +178,17 @@ function minify(files, options) {
             toplevel = toplevel[action](option);
             files[toplevel.start.file] = toplevel.print_to_string().replace(orig, "");
         });
+        if (options.validate) toplevel.validate_ast();
         if (timings) timings.rename = Date.now();
         if (options.rename) {
             toplevel.figure_out_scope(options.mangle);
             toplevel.expand_names(options.mangle);
         }
         if (timings) timings.compress = Date.now();
-        if (options.compress) toplevel = new Compressor(options.compress).compress(toplevel);
+        if (options.compress) {
+            toplevel = new Compressor(options.compress).compress(toplevel);
+            if (options.validate) toplevel.validate_ast();
```

Comment:
I thought having a check after each compress pass would have some merit, but I suppose this is good enough.

Reply:
I thought about it, but this seems like a better boundary 😅

Reply:
I've seen similar invalid AST subtrees disappear with an additional compress pass. But there are a number of reminify variants so that helps.